### DISCUSSION TITLE
Unable to compile podman with rootless true by default

### DISCUSSION THREAD

**Discussion by ntsbtz** [containers/podman#15440] [General]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I am trying to build podman on PhotonOS but podman rootless is not enable by default. podman info is showing rootless as false.

This is the build section to build an podman RPM :

%build
%if "%{_vendor}" != "debbuild"
%set_build_flags
export CGO_CFLAGS=$CFLAGS
CGO_CFLAGS=$(echo $CGO_CFLAGS | sed 's/-flto=auto//g')
CGO_CFLAGS=$(echo $CGO_CFLAGS | sed 's/-Wp,D_GLIBCXX_ASSERTIONS//g')
CGO_CFLAGS=$(echo $CGO_CFLAGS | sed 's/-specs=\/usr\/lib\/rpm\/redhat\/redhat-annobin-cc1//g')

%ifarch x86_64
export CGO_CFLAGS+=" -m64 -mtune=generic -fcf-protection=full"
%endif
%endif

export GO111MODULE=off
export GOPATH=$(pwd)/_build:$(pwd)

mkdir _build
cd _build
mkdir -p src/%{provider}.%{provider_tld}/%{project}
ln -s ../../../../ src/%{import_path}
cd ..
ln -s vendor src

LDFLAGS="-X %{import_path}/libpod/define.buildInfo=$(date +%s)"

%gobuild -o bin/rootlessport %{import_path}/cmd/rootlessport

export BUILDTAGS="seccomp exclude_graphdriver_devicemapper $(hack/btrfs_installed_tag.sh) $(hack/btrfs_tag.sh) $(hack/libdm_tag.sh) $(hack/selinux_tag.sh) $(hack/systemd_tag.sh) $(hack/libsubid_tag.sh)"
%gobuild -o bin/%{name} %{import_path}/cmd/%{name}

export BUILDTAGS="seccomp exclude_graphdriver_devicemapper exclude_graphdriver_btrfs btrfs_noversion $(hack/selinux_tag.sh) $(hack/systemd_tag.sh) $(hack/libsubid_tag.sh) remote"
%gobuild -o bin/%{name}-remote %{import_path}/cmd/%{name}

**Steps to reproduce the issue:**

1.Building podman RPM

2.

3.

**Describe the results you received:**
rootless: false

**Describe the results you expected:**
rootless: true 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.1.1
API Version:  4.1.1
Go Version:   go1.18.1
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.3-1.ph4.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.3, commit: '
  cpuUtilization:
    idlePercent: 99.99
    systemPercent: 0
    userPercent: 0.01
  cpus: 16
  distribution:
    distribution: photon
    version: "4.0"
  eventLogger: journald
  hostname: testvm
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.10.118-14.ph4
  linkmode: dynamic
  logDriver: journald
  memFree: 3275042816
  memTotal: 4130152448
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.1-3.ph4.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.1
      spec: 1.0.2-dev
      go: go1.18.1
      libseccomp: 2.5.0
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-1.ph4.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.0
  swapFree: 0
  swapTotal: 0
  uptime: 20h 56m 15.19s (Approximately 0.83 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries: {}
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 33620226048
  graphRootUsed: 2038566912
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.1.1
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.18.1
  Os: linux
  OsArch: linux/amd64
  Version: 4.1.1

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.1.1-1.ph4.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/No

**Additional environment details (AWS, VirtualBox, physical, etc.):**
I am building the podman on PhotonOS

**Comment by mheon** [comment]
There are no build tags to enable or disable rootless. Rootless mode is automatically enabled when you run Podman as a user who is not root. If you run Podman as root, rootless mode will never be true, even if the binary supports it. I suspect you're running Podman as root.

**Comment by ntsbtz** [comment]
> 
Thanks for quick response @mheon that is what i was also assuming there is no tag to enable or disable rootless. Further if i installed podman and i am a root user podman info will show rootless as false as you said right ?

Now i created another user and login to that 
1. ssh rootlessUsr@localhost 
2. podman info should show here rootless as true.  Please correct me if i am wrong 

Now podman info command throwing below error :

Error: kernel does not support overlay fs: 'overlay' is not supported over extfs at "/home/rootlessVmware/.local/share/containers/storage/overlay": backing file system is unsupported for this graph driver

**Comment by mheon** [comment]
You most likely do not have a new enough kernel for rootless overlayfs and do not have fuse-overlayfs installed.

**Reply by ntsbtz** [reply]
Hi @mheon,  I am able to pull the images with rootless user but while running the container with below inputs facing some error please check the below detail logs for more info:

podman run -dt -p 8080:80/tcp docker.io/library/httpd
Error: OCI runtime error: runc: runc create failed: unable to start container process: chdir to cwd ("/usr/local/apache2") set in config.json failed: operation not supported

**Comment by rhatdan** [comment]
This is a discussion not an issue so converting.

**Comment by ntsbtz** [comment]
Log info for same :

podman --log-level=debug run -dt -p 8080:80/tcp docker.io/library/httpd

INFO[0000] podman filtering at log level debug          
DEBU[0000] Called run.PersistentPreRunE(podman --log-level=debug run -dt -p 8080:80/tcp docker.io/library/httpd) 
DEBU[0000] Merged system config "/usr/share/containers/containers.conf" 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/rootlessVmware/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding run root "/run/user/10000" with "/run/user/1000/containers" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/rootlessVmware/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/1000/containers     
DEBU[0000] Using static dir /home/rootlessVmware/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/1000/libpod/tmp      
DEBU[0000] Using volume path /home/rootlessVmware/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] overlay: mount_program=/usr/bin/fuse-overlayfs 
DEBU[0000] overlay: mount_program=/usr/bin/fuse-overlayfs 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=false, usingMetacopy=false 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 49             
DEBU[0000] Failed to add podman to systemd sandbox cgroup: dial unix /run/user/0/bus: connect: permission denied 
DEBU[0000] Adding port mapping from 8080 to 80 length 1 protocol "tcp" 
DEBU[0000] Pulling image docker.io/library/httpd (policy: missing) 
DEBU[0000] Looking up image "docker.io/library/httpd" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "docker.io/library/httpd:latest" ...  
DEBU[0000] parsed reference into "[overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] Found image "docker.io/library/httpd" as "docker.io/library/httpd:latest" in local containers storage 
DEBU[0000] Found image "docker.io/library/httpd" as "docker.io/library/httpd:latest" in local containers storage ([overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825) 
DEBU[0000] Looking up image "docker.io/library/httpd:latest" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "docker.io/library/httpd:latest" ...  
DEBU[0000] parsed reference into "[overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] Found image "docker.io/library/httpd:latest" as "docker.io/library/httpd:latest" in local containers storage 
DEBU[0000] Found image "docker.io/library/httpd:latest" as "docker.io/library/httpd:latest" in local containers storage ([overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825) 
DEBU[0000] Looking up image "docker.io/library/httpd" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "docker.io/library/httpd:latest" ...  
DEBU[0000] parsed reference into "[overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] Found image "docker.io/library/httpd" as "docker.io/library/httpd:latest" in local containers storage 
DEBU[0000] Found image "docker.io/library/httpd" as "docker.io/library/httpd:latest" in local containers storage ([overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825) 
DEBU[0000] Inspecting image a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825 
DEBU[0000] exporting opaque data as blob "sha256:a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] exporting opaque data as blob "sha256:a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] exporting opaque data as blob "sha256:a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] Inspecting image a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825 
DEBU[0000] Inspecting image a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825 
DEBU[0000] Inspecting image a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825 
DEBU[0000] using systemd mode: false                    
DEBU[0000] No hostname set; container's hostname will default to runtime default 
DEBU[0000] Loading seccomp profile from "/usr/share/containers/seccomp.json" 
DEBU[0000] Allocated lock 0 for container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 
DEBU[0000] parsed reference into "[overlay@/home/rootlessVmware/.local/share/containers/storage+/run/user/1000/containers:overlay.mount_program=/usr/bin/fuse-overlayfs,overlay.mount_program=/usr/bin/fuse-overlayfs]@a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] exporting opaque data as blob "sha256:a981c8992512d65c9b450a9ecabb1cb9d35bb6b03f3640f86471032d5800d825" 
DEBU[0000] Created container "728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60" 
DEBU[0000] Container "728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60" has work directory "/home/rootlessVmware/.local/share/containers/storage/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata" 
DEBU[0000] Container "728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60" has run directory "/run/user/1000/containers/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata" 
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] overlay: mount_program=/usr/bin/fuse-overlayfs 
DEBU[0000] overlay: mount_program=/usr/bin/fuse-overlayfs 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=false, usingMetacopy=false 
DEBU[0000] Cached value indicated that overlay is not supported 
DEBU[0000] Check for idmapped mounts support            
DEBU[0000] overlay: mount_data=lowerdir=/home/rootlessVmware/.local/share/containers/storage/overlay/l/JE4FIK2H6AJQMIOXRURT2OPDJX:/home/rootlessVmware/.local/share/containers/storage/overlay/l/R4DT4OCIEHP53VDI7MSWW74FNP:/home/rootlessVmware/.local/share/containers/storage/overlay/l/57YTNK3FDWFGKBHQJLJZJARW4G:/home/rootlessVmware/.local/share/containers/storage/overlay/l/OAOYOZKE5SZKDFSL75T5NKGG5B:/home/rootlessVmware/.local/share/containers/storage/overlay/l/U7POTPERVDRIOLXCRNYKX4AKH2,upperdir=/home/rootlessVmware/.local/share/containers/storage/overlay/594e6ebd746afcd0746ebf2687434dde91ee3cd51dd7335acfdeff81f0e64dea/diff,workdir=/home/rootlessVmware/.local/share/containers/storage/overlay/594e6ebd746afcd0746ebf2687434dde91ee3cd51dd7335acfdeff81f0e64dea/work, 
DEBU[0000] Made network namespace at /run/user/1000/netns/netns-76580c1a-d67a-1d29-3a0c-ed1cce7d900b for container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 
DEBU[0000] slirp4netns command: /usr/bin/slirp4netns --disable-host-loopback --mtu=65520 --enable-sandbox --enable-seccomp --enable-ipv6 -c -e 3 -r 4 --netns-type=path /run/user/1000/netns/netns-76580c1a-d67a-1d29-3a0c-ed1cce7d900b tap0 
DEBU[0000] Mounted container "728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60" at "/home/rootlessVmware/.local/share/containers/storage/overlay/594e6ebd746afcd0746ebf2687434dde91ee3cd51dd7335acfdeff81f0e64dea/merged" 
DEBU[0000] Created root filesystem for container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 at /home/rootlessVmware/.local/share/containers/storage/overlay/594e6ebd746afcd0746ebf2687434dde91ee3cd51dd7335acfdeff81f0e64dea/merged 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="Starting parent driver" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="opaque=map[builtin.readypipepath:/run/user/1000/libpod/tmp/rootlessport356056423/.bp-ready.pipe builtin.socketpath:/run/user/1000/libpod/tmp/rootlessport356056423/.bp.sock]" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="Starting child driver in child netns (\"/proc/self/exe\" [rootlessport-child])" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="Waiting for initComplete" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="initComplete is closed; parent and child established the communication channel" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg="Exposing ports [{ 80 8080 1 tcp}]" 
DEBU[0000] rootlessport: time="2022-08-29T18:38:05Z" level=info msg=Ready 
DEBU[0000] rootlessport is ready                        
DEBU[0000] /etc/system-fips does not exist on host, not mounting FIPS mode subscription 
DEBU[0000] reading hooks from /usr/share/containers/oci/hooks.d 
DEBU[0000] Workdir "/usr/local/apache2" resolved to host path "/home/rootlessVmware/.local/share/containers/storage/overlay/594e6ebd746afcd0746ebf2687434dde91ee3cd51dd7335acfdeff81f0e64dea/merged/usr/local/apache2" 
DEBU[0000] Created OCI spec for container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 at /home/rootlessVmware/.local/share/containers/storage/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata/config.json 
DEBU[0000] /usr/bin/conmon messages will be logged to syslog 
DEBU[0000] running conmon: /usr/bin/conmon               args="[--api-version 1 -c 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 -u 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 -r /usr/bin/runc -b /home/rootlessVmware/.local/share/containers/storage/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata -p /run/user/1000/containers/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata/pidfile -n thirsty_nobel --exit-dir /run/user/1000/libpod/tmp/exits --full-attach -l journald --log-level debug --syslog -t --conmon-pidfile /run/user/1000/containers/overlay-containers/728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /home/rootlessVmware/.local/share/containers/storage --exit-command-arg --runroot --exit-command-arg /run/user/1000/containers --exit-command-arg --log-level --exit-command-arg debug --exit-command-arg --cgroup-manager --exit-command-arg cgroupfs --exit-command-arg --tmpdir --exit-command-arg /run/user/1000/libpod/tmp --exit-command-arg --network-config-dir --exit-command-arg  --exit-command-arg --network-backend --exit-command-arg cni --exit-command-arg --volumepath --exit-command-arg /home/rootlessVmware/.local/share/containers/storage/volumes --exit-command-arg --runtime --exit-command-arg runc --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg --storage-opt --exit-command-arg overlay.mount_program=/usr/bin/fuse-overlayfs --exit-command-arg --storage-opt --exit-command-arg overlay.mount_program=/usr/bin/fuse-overlayfs --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg --syslog --exit-command-arg container --exit-command-arg cleanup --exit-command-arg 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60]"
INFO[0000] Failed to add conmon to cgroupfs sandbox cgroup: error creating cgroup for cpu: mkdir /sys/fs/cgroup/cpu/conmon: permission denied 
DEBU[0000] Received: -1                                 
DEBU[0000] Cleaning up container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 
DEBU[0000] Tearing down network namespace at /run/user/1000/netns/netns-76580c1a-d67a-1d29-3a0c-ed1cce7d900b for container 728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60 
DEBU[0000] Unmounted container "728773e25961379aac542422474b3de891c037d21a66144acb482250afd0fd60" 
DEBU[0000] ExitCode msg: "runc: time=\"2022-08-29t18:38:06z\" level=warning msg=\"unable to get oom kill count\" error=\"no directory specified for memory.oom_control\"\ntime=\"2022-08-29t18:38:06z\" level=error msg=\"runc create failed: unable to start container process: chdir to cwd (\\\"/usr/local/apache2\\\") set in config.json failed: operation not supported\": oci runtime error" 
Error: OCI runtime error: runc: time="2022-08-29T18:38:06Z" level=warning msg="unable to get oom kill count" error="no directory specified for memory.oom_control"
time="2022-08-29T18:38:06Z" level=error msg="runc create failed: unable to start container process: chdir to cwd (\"/usr/local/apache2\") set in config.json failed: operation not supported"
